calcom/cal.com
# feat: Alby integration

## What does this PR do?



Alby Implementation for https://github.com/calcom/cal.com/pull/11347

This enables Cal.com users to make setup paid meetings and accept sats from their guests. Guests must pay in advance to book the meeting.

Features:
- Connect with your existing Alby account with or create a new one with a few clicks
- Pay for meetings by scanning, clicking, copying, or with a WebLN extension
- Sats - USD conversion on hover

This PR also introduces some additional functionality which needs to be reviewed:
- support for custom currencies / currency symbols
- App cards now display if an app is setup correctly and has a link to the setup page
- Add a way to navigate between setup page and app store entry
- Payment webhook handling extracted into a new lib function
- Replaced getStaticProps with getServerSideProps on app setup pages in order to access app credentials (maybe there is a better way and this change can be undone)



https://www.loom.com/share/75dfc1d394bf4756a4520523afb913fd?sid=37430206-7e7d-4f85-af37-a454f144c9cc

## Requirement/Documentation



- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

N/A

## Type of change




## How should this be tested?



Manually

- Are there environment variables that should be set? yes: Alby OAuth client ID and secret (see .env.example)
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)? Booking successfully completed after payment
- Any other important info that could help to test that PR: use a tunnel (e.g. ngrok) so that payment webhooks work

## Mandatory Tasks


## Checklist



- My code doesn't follow the style guidelines of this project
- I haven't added tests that prove my fix is effective or that my feature works
